pypose/module/cameras.py

```diff
@@ -0,0 +1,149 @@
+from typing import Optional
+
+import pypose
+import torch.nn
+
+
+class CamerasBase(torch.nn.Module):
+    """
+    `CamerasBase` implements a base class for all cameras.
+    For cameras, there are two different coordinate systems (or spaces)
+    - World coordinate system: This is the system the object lives - the world.
+    - Camera view coordinate system: This is the system that has its origin on
+        the camera and the Z-axis perpendicular to the image plane.
+        We assume that +X points left, and +Y points up and
+        +Z points out from the image plane.
+        The transformation from world --> view happens after applying a rotation (R)
+        and translation (T)
+    CameraBase defines methods that are common to all camera models:
+        - `camera_center` that returns the optical center of the camera in
+            world coordinates
+        - `world_to_view_transform` which returns a 3D transform from
+            world coordinates to the camera view coordinates (R, T)
+        - `full_projection_transform` which composes the projection
+            transform (P) with the world-to-view transform (R, T)
+
+    Args:
+        pose (LieTensor): A tensor of shape (B, *) where B is the batch size.
+    """
+
+    def __init__(self, pose=None, ):
+        super().__init__()
+        if pose is not None:
+            self.pose = pose
+        elif self.pose is None:
+            self.pose = pypose.identity_SE3()
+
+    def get_camera_center(self):
+        """
+        Returns the camera center in world coordinates.
+        Returns:
+            A tensor of shape (B, 3) where B is the batch size.
+        """
+        return pypose.Inv(self.pose)[..., :3]
+
+    def world_to_view_transform(self, points):
```

Comment:
How about `def world2camera(self, points):`
this will follow our [naming convention](https://pypose.org/docs/preview/pr/184/convert/).